library/std/src/os/unix/net/tests.rs

```diff
@@ -638,3 +630,91 @@ fn test_send_vectored_with_ancillary_unix_datagram() {
         unreachable!("must be ScmRights");
     }
 }
+
+#[cfg(any(target_os = "android", target_os = "emscripten", target_os = "linux"))]
+#[test]
+fn test_send_vectored_with_ancillary_to_udp_socket_ttl() {
+    let socket1 = or_panic!(UdpSocket::bind("127.0.0.1:0"));
+    let socket2 = or_panic!(UdpSocket::bind("127.0.0.1:0"));
+
+    let addr1 = or_panic!(socket1.local_addr());
+    let addr2 = or_panic!(socket2.local_addr());
+
+    or_panic!(socket2.set_recvttl(true));
+
+    let buf1 = [1; 8];
+    let bufs_send = &[IoSlice::new(&buf1[..])][..];
+
+    let mut ancillary1_buffer = [0; 64];
+    let mut ancillary1 = IpAncillary::new(&mut ancillary1_buffer[..]);
+    assert!(ancillary1.add_ttl(20));
+
+    let usize =
+        or_panic!(socket1.send_vectored_with_ancillary_to(&bufs_send, &mut ancillary1, &addr2));
+    assert_eq!(usize, 8);
+
+    let mut buf2 = [0; 8];
+    let mut bufs_recv = &mut [IoSliceMut::new(&mut buf2[..])][..];
+
+    let mut ancillary2_buffer = [0; 64];
+    let mut ancillary2 = IpAncillary::new(&mut ancillary2_buffer[..]);
+
+    let (usize, truncated, addr) =
+        or_panic!(socket2.recv_vectored_with_ancillary_from(&mut bufs_recv, &mut ancillary2));
+    assert_eq!(usize, 8);
+    assert_eq!(truncated, false);
+    assert_eq!(addr1, addr);
+    assert_eq!(buf1, buf2);
+
+    let mut ancillary_data_vec = Vec::from_iter(ancillary2.messages());
+    assert_eq!(ancillary_data_vec.len(), 1);
+    if let IpAncillaryData::Ttl(ttl) = ancillary_data_vec.pop().unwrap().unwrap() {
+        assert_eq!(ttl, 20);
+    } else {
+        unreachable!("must be Ttl");
+    }
```

Comment:
this could be rewritten using [matches!](https://doc.rust-lang.org/std/macro.matches.html)

Reply:
I did not know this macro. Thanks